capstone/src/main/java/com/google/sps/Objects/response/MemberResponse.java

```diff
@@ -58,6 +58,30 @@ public static MemberResponse fromEntity(Entity entity, boolean includeBadges) {
     return new MemberResponse(profilePic, firstName, lastName, badges, userId);
   }
 
+  /**
+   * Checks if two Member Response Objects are equal
+   *
+   * @param other Member Response object 
+   */
+  @Override
+  public boolean equals(Object other) {
+    if (other == null) return false;
+    if (other == this) return true;
+    if (!(other instanceof MemberResponse)) return false;
+    MemberResponse memResponse = (MemberResponse) other;
+    if (memResponse.badges != null && memResponse.badges.size() > 0) {
+      return profilePic.equals(memResponse.profilePic) &&
+        firstName.equals(memResponse.firstName) &&
+        lastName.equals(memResponse.lastName) &&
+        userId.equals(memResponse.userId) &&
+        badges.containsAll(memResponse.badges);
+    }
+    return profilePic.equals(memResponse.profilePic) &&
+      firstName.equals(memResponse.firstName) &&
+      lastName.equals(memResponse.lastName) &&
+      userId.equals(memResponse.userId);
```

Comment:
Could you refactor this so you're only checking profilePic, firstName, lastName, userId in one place in this function so that it's easier to maintain when you add/remove fields?

Reply:
sure! changed